### PR TITLE
LOOP-885 Increase Loop Core Data cache expiration duration

### PR DESCRIPTION
**Pull request by darinkrauss**
- https://tidepool.atlassian.net/browse/LOOP-885
- Add global local cache duration configuration
- Add feature flags for scenarios and simulated core data
- Add debug menu for scenarios and simulated core data
- Allow purging historical core data
- Allow generating simulated historical core data

Note: This PR requires @ps2 and at least one other reviewer.

### CODE REVIEW

**Comment by darinkrauss** [comment]
LoopWorkspace build PR - https://github.com/tidepool-org/LoopWorkspace/pull/108

File: Loop.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -280,7 +280,6 @@
 		4F70C2101DE8FAC5006380B7 /* StatusExtensionDataManager.swift in Sources */ = {isa = PBXBuildFile; fileRef = 4F70C20F1DE8FAC5006380B7 /* StatusExtensionDataManager.swift */; };
 		4F70C2121DE900EA006380B7 /* StatusExtensionContext.swift in Sources */ = {isa = PBXBuildFile; fileRef = 4F70C2111DE900EA006380B7 /* StatusExtensionContext.swift */; };
 		4F70C2131DE90339006380B7 /* StatusExtensionContext.swift in Sources */ = {isa = PBXBuildFile; fileRef = 4F70C2111DE900EA006380B7 /* StatusExtensionContext.swift */; };
-		4F73F5FC20E2E7FA00E8D82C /* GlucoseStore.swift in Sources */ = {isa = PBXBuildFile; fileRef = 4F73F5FB20E2E7FA00E8D82C /* GlucoseStore.swift */; };
```
Comment (rickpasetto) [inline Loop.xcodeproj/project.pbxproj:283]: Why was `GlucoseStore` removed here? Looks like it is the watch app version of the file...was this intentional?

**Review by rickpasetto — APPROVED** [review]
LGTM 🧀

File: Loop.xcodeproj/project.pbxproj (hunk 1 above)
Comment (darinkrauss) [reply Loop.xcodeproj/project.pbxproj:283]: The file only had one empty extension. Either way, it was no longer needed.

**Review by ps2 — APPROVED** [review]
LGTM!

**Comment by darinkrauss** [comment]
@ps2 @rickpasetto Per Pete's suggestions, I refactored much of this work to take all of the simulated core data out of LoopKit and the main store implementations, leaving only a straightforward "bulk add" API. Would you please review both PRs? Thanks!

**Review by rickpasetto — APPROVED** [review]
LGTM 🐟

**Review by ps2 — APPROVED** [review]
LGTM!

**Pull request merged**